test/transmuxer-worker.test.js

```diff
@@ -0,0 +1,740 @@
+import QUnit from 'qunit';
+import TransmuxWorker from 'worker!../src/transmuxer-worker.worker.js';
+import {
+  muxed as muxedSegment,
+  caption as captionSegment
+} from './test-segments';
+// needed for plugin registration
+import '../src/videojs-http-streaming';
+
+const createTransmuxer = (isPartial) => {
+  const transmuxer = new TransmuxWorker();
+
+  transmuxer.postMessage({
+    action: 'init',
+    options: {
+      remux: false,
+      keepOriginalTimestamps: true,
+      handlePartialData: isPartial
+    }
+  });
+
+  return transmuxer;
+};
+
+// The final done message from the Transmux worker
+// will have type `transmuxed`
+const isFinalDone = (event) => {
+  return event.data.action === 'done' &&
+    event.data.type === 'transmuxed';
+};
+
+QUnit.module('Transmuxer Worker', {
+  beforeEach(assert) {
+    this.transmuxer = null;
```

Comment:
this is just to  make sure `this.transmuxer` is defined, in case one of the tests doesn't use the regular `this.transmuxer =` method, so the afterEach can terminate the transmuxer properly

Reply:
Does it change anything though? If a test doesn't use the regular method, wouldn't it still not terminate?

Reply:
Sorry, I conflated two things I was trying to say. 
i. if using the regular method, this ensures we don't leave web workers lying around
ii. if a test happens to not define `this.transmuxer`, then we don't get errors for no good reason